### PR TITLE
Optimize DTR fetching

### PR DESCRIPTION
**Pull request by bhollis**
With a batch size of 10, I was requiring 48 serial requests to DTR to load ratings, which took forever. This PR optimizes a few things:

1. Bump the batch size to 100.
2. Load all the batches in parallel, instead of serially.
3. Optimize the calculation of which items are new.
4. Include "roll" in whether an item is new, not just item hash.
5. Skip doing expensive work when the response is empty.

### CODE REVIEW

**Comment by 48klocs** [comment]
Some of this is a no go - the DTR folks requested that we fetch 10 max at a time, with at most two requests per person coming through at a time.

I’ll give the rest a look tomorrow.

**Comment by bhollis** [comment]
If that's the case, then we need to implement local indexedDB caching of reviews. 48 requests is just too much.

File: src/app/destinyTrackerApi/d2-itemListBuilder.ts [hunk 1]
```diff
@@ -49,20 +49,24 @@ export function getVendorItemList(
...
 function getNewVendorItems(vendorItems: D2ItemFetchRequest[], reviewDataCache: D2ReviewDataCache) {
-  const allKnownDtrItems = reviewDataCache.getItemStores();
+  const allKnownDtrItems = new Set(
+    reviewDataCache.getItemStores().map((kdi) => `${kdi.referenceId}-${kdi.roll}`)
+  );
 
-  const unmatched = vendorItems.filter((vi) =>
-    allKnownDtrItems.every((di) => di.referenceId !== vi.referenceId)
+  const unmatched = vendorItems.filter(
+    (di) => !allKnownDtrItems.has(`${di.referenceId}-${getD2Roll(di.availablePerks)}`)
```
Comment (48klocs) [inline src/app/destinyTrackerApi/d2-itemListBuilder.ts]: We build this string independently here and in the reducer.

Probably makes sense to make a single function that computes + returns the string alongside `getD2Roll`.

**Comment by 48klocs** [comment]
The changes outside of the fetcher class look good (but I think we want another function to get the `D2Item`->`string` key transform because it now exists in a number of separate places).

The changes inside of the fetcher are troublesome - I asked the DTR folks about revisiting the number of items we can send in a single request. We have other changes coming sooner or later, so we'll see.

**Comment by bhollis** [comment]
OK. How does this look?

**Comment by 48klocs** [comment]
Looks good!

Feel free to merge on in whenever (I didn't merge it because it might be getting close to release time).

**Pull request merged**